From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20Offenh=C3=A4user?= <offenhaeuser@protonmail.com>
Date: Tue, 7 Feb 2023 21:25:16 +0100
Subject: [PATCH] Read the executable path from /proc/self/exe

When looking for the executable path, we behave exactly like Linux.
---
 src/openrct2/platform/Platform.Linux.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/openrct2/platform/Platform.Linux.cpp b/src/openrct2/platform/Platform.Linux.cpp
index d8c48bcd3df5c75c6f9f5a87288d88cb04df872a..ac5357dba07442a508c71e52a9636d06b79acc6e 100644
--- a/src/openrct2/platform/Platform.Linux.cpp
+++ b/src/openrct2/platform/Platform.Linux.cpp
@@ -182,7 +182,7 @@ namespace OpenRCT2::Platform
     std::string GetCurrentExecutablePath()
     {
         char exePath[PATH_MAX] = { 0 };
-    #ifdef __linux__
+    #if defined(__linux__) || defined(__serenity__)
         auto bytesRead = readlink("/proc/self/exe", exePath, sizeof(exePath));
         if (bytesRead == -1)
         {
